client/cli/src/runner.rs

```diff
@@ -172,36 +172,29 @@ impl<C: SubstrateCli> Runner<C> {
 		info!("⛓  Native runtime: {}", runtime_version);
 	}
 
-	/// A helper function that runs an `AbstractService` with tokio and stops if the process
+	/// A helper function that runs a node with tokio and stops if the process
 	/// receives the signal `SIGTERM` or `SIGINT`. It can run a full or a light node depending on
 	/// the node's configuration.
-	pub fn run_node<SL, SF>(
+	pub fn run_node(
 		self,
-		new_light: impl FnOnce(Configuration) -> sc_service::error::Result<SL>,
-		new_full: impl FnOnce(Configuration) -> sc_service::error::Result<SF>,
+		new_light: impl FnOnce(Configuration) -> sc_service::error::Result<TaskManager>,
+		new_full: impl FnOnce(Configuration) -> sc_service::error::Result<TaskManager>,
```

Comment:
Maybe this is far-fetched but I wonder if we still need 2 closures??

The reason why we had 2 closures was because both closures return a different impl of AbstractService so Rust refused to compile (because their types are different). But now that they both return a TaskManager (which is not a trait), I believe we don't have this issue anymore.

(I saw this 2-closures pattern elsewhere in the code of this PR, maybe it is there for the same reason.)

Reply:
If that's the case, I think you can remove all the `run_node*` methods from sc-cli's API. In the bins you will need to use `.run_service_until_exit()` instead (and do a `match` for the light/full node).

Reply:
Done : )